app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java

```diff
@@ -0,0 +1,96 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigSettings;
+
+import org.mozilla.focus.BuildConfig;
+import org.mozilla.focus.R;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+
+/**
+ * Provide helper for Firebase functionality
+ */
+public class FirebaseHelper {
+
+    public static final String RATE_APP_DIALOG_TEXT_TITLE = "rate_app_dialog_text_title";
+    public static final String RATE_APP_DIALOG_TEXT_CONTENT = "rate_app_dialog_text_content";
+    private static final String TAG = "FirebaseHelper";
+
+    private static WeakReference<FirebaseRemoteConfig> REMOTE_CONFIG;
+    private static long CACHE_EXPIRATION = 3600; // 1 hour in seconds.
+
+    public static String getString(String key) {
+        final FirebaseRemoteConfig config = REMOTE_CONFIG.get();
+        if (config != null) {
+            return config.getString(key);
+        }
+        return null;
+    }
+
+    public static void init(final Context context) {
+        new Thread(new Runnable() {
+            public void run() {
+                internalInit(context);
```

Comment:
Why put it to worker thread? Could there be issues to initial firebase in non-UI thread? For example, what if we access the remote config object in UI thread, however, the initializing code is not executed yet?